Troubleshooting.md

```diff
@@ -51,3 +51,59 @@ there are two possible reasons for this:
 2. Firebase iOS SDK also swizzles AppDelegate methods when configured on your iOS app. If after disabling notifee,
    if the delegates are still not called, this may be the reason. Method swizzling can be turned off by following
    [Firebase's documentation](https://firebase.google.com/docs/cloud-messaging/ios/client).
+
+### Deep links in push notifications not getting sent over to React Native layer when the iOS app is terminated
+
+When the iOS app is terminated (killed from the app switcher by swiping up), there is a bug in React Native that 
+prevents the native layer from calling the listeners set up in React Native to listen to incoming deep links. 
+The listener in question here is `Linking.getInitialURL()` which is expected to be called when a deep link is 
+available while the app is terminated.
+
+You can find the open issue describing this problem on React Native's GitHub repository [Here](https://github.com/facebook/react-native/issues/32350). 
+There are many workarounds suggested in this issue's thread and a few other similar issues. 
+However, the workaround that worked for us involves intercepting the launch arguments in the app delegate and adding 
+a key `UIApplicationLaunchOptionsURLKey`, which React Native expects to be present when calling the `Linking.getInitialURL()` listener.
+
+
+Here's a method to implement this workaround:
+
+
+```swift 
+- (NSMutableDictionary *)getLaunchOptionsWithURL:(NSDictionary * _Nullable)launchOptions {
+  NSMutableDictionary *launchOptionsWithURL = [NSMutableDictionary dictionaryWithDictionary:launchOptions];
+  if (launchOptions[UIApplicationLaunchOptionsRemoteNotificationKey]) {
+    NSDictionary *remoteNotification = launchOptions[UIApplicationLaunchOptionsRemoteNotificationKey];
+    
+    if (remoteNotification[@"url"]) {
+      NSString *initialURL = remoteNotification[@"url"];
+      if (!launchOptions[UIApplicationLaunchOptionsURLKey]) {
+        launchOptionsWithURL[UIApplicationLaunchOptionsURLKey] = [NSURL URLWithString:initialURL];
```

Comment:
So the RN linking code expects this key to be set, but it isn't set by default by iOS when tapping a notification?

Reply:
Yeah, so there are a bunch of things going on here, and I don't fully understand everything going on BTS, but here's what I know - 

1. [`getInitialUrl`](https://github.com/facebook/react-native/blob/8c53ac60782d8ccd2ccd7706d41e60cda93e2988/packages/react-native/Libraries/LinkingIOS/RCTLinkingManager.mm#L154) is the listener that is used on react native and it depends on the launch options key `UIApplicationLaunchOptionsURLKey`. This key is present when the app is invoked from a non-push deep link (like from an email/SMS) but absent when the app is launched (cold start) from a push notification in which case the launch argument contains the push payload within the key `UIApplicationLaunchOptionsRemoteNotificationKey`. Since `getInitialUrl` looks up the key `UIApplicationLaunchOptionsURLKey` this workaround moves the deep link url in the push payload to the top level of launch arguments where `getInitialUrl` will find it and the RN listeners will get the deep link value. I think this is a case where Apple being inconsistent with deep linking (push vs non-push) is an issue but the workaround I added could very well be added to the `getInitialUrl` method on RN to fix this issue.
2. The above point was when the app is terminated, but in all other cases, the other listener `Linking.addEventListener` is used and this is called whenever the app is in the foreground/background by the `RCTLinkingManager`. Why this is not used in all cases is not clear to me, if I had to take a wild guess it's got something to do with the RN run time not being initialized during cold start.

Reply:
@evan-masseau  Thanks for asking the question. Made me dig a little deeper.

Reply:
Awesome, that makes a lot of sense!